ydb/library/yql/minikql/comp_nodes/ut/mkql_block_map_join_ut.cpp

```diff
@@ -665,6 +684,322 @@ Y_UNIT_TEST_SUITE(TMiniKQLBlockMapJoinBasicTest) {
+            const auto& found = rightMap.find(*keyInit[i]);
+            if (found != rightMap.cend()) {
+                keyExpected.push_back(keyInit[i]);
+                subkeyExpected.push_back(subkeyInit[i]);
+                valueExpected.push_back(valueInit[i]);
+                rightExpected.push_back(found->second);
+            }
+        }
+        // 4. Convert input and expected TVectors to List<UV>.
+        const auto [leftType, leftList] = ConvertVectorsToTuples(setup,
+            keyInit, subkeyInit, valueInit);
+        const auto [expectedType, expected] = ConvertVectorsToTuples(setup,
+            keyExpected, subkeyExpected, valueExpected, rightExpected);
+        // 5. Build "right" computation node.
+        const auto rightKeys = BuildListNodes(pgmBuilder, rightKeyInit);
+        const auto rightPayloads = BuildListNodes(pgmBuilder, rightPayloadInit);
+        const auto rightMapNode = MakeDict(pgmBuilder, rightKeys, rightPayloads);
+        // 6. Run tests.
+        RunTestBlockJoin(setup, EJoinKind::Inner, expectedType, expected,
+                         rightMapNode, leftType, leftList, {0});
+    }
+
+    Y_UNIT_TEST(TestInnerMultiOnOptionalUint64) {
+        TSetup<false> setup;
+        TProgramBuilder& pgmBuilder = *setup.PgmBuilder;
+        // 1. Make input for the "left" stream.
+        TVector<std::optional<ui64>> keyInit(testSize);
+        std::iota(keyInit.begin(), keyInit.end(), 1);
+        TVector<ui64> subkeyInit;
+        std::transform(keyInit.cbegin(), keyInit.cend(), std::back_inserter(subkeyInit),
+            [](const auto key) { return *key * 1001; });
+        TVector<TString> valueInit;
+        std::transform(keyInit.cbegin(), keyInit.cend(), std::back_inserter(valueInit),
+            [](const auto key) { return threeLetterValues[*key]; });
+        // 1a. Make some keys NULL
+        keyInit[0] = std::nullopt;
+        // 2. Make input for the "right" dict.
+        const TVector<ui64> rightKeyInit(fibonacci.cbegin(), fibonacci.cend());
+        TVector<TString> rightPayload1Init;
+        std::transform(rightKeyInit.cbegin(), rightKeyInit.cend(), std::back_inserter(rightPayload1Init),
+            [](const auto key) { return std::to_string(key); });
+        TVector<TString> rightPayload2Init;
+        std::transform(rightKeyInit.cbegin(), rightKeyInit.cend(), std::back_inserter(rightPayload2Init),
+            [](const auto key) { return std::to_string(key * 1001); });
+        // 3. Make "expected" data.
+        TMap<ui64, TVector<TString>> rightMultiMap;
+        for (size_t i = 0; i < rightKeyInit.size(); i++) {
+            rightMultiMap[rightKeyInit[i]] = {rightPayload1Init[i], rightPayload2Init[i]};
+        }
+        TVector<std::optional<ui64>> keyExpected;
+        TVector<ui64> subkeyExpected;
+        TVector<TString> valueExpected;
+        TVector<TString> rightExpected;
+        for (size_t i = 0; i < keyInit.size(); i++) {
+            if (!keyInit[i]) continue;
```

Comment:
Typo: codestyle violation, please use braces and split this operator into several lines.